py-polars/polars/internals/dataframe/frame.py

```diff
@@ -5658,7 +5659,12 @@ def select(
 
     def with_columns(
         self,
-        exprs: pli.Expr | pli.Series | Sequence[pli.Expr | pli.Series] | None = None,
+        exprs: (
+            PolarsExprType
+            | PythonLiteral
+            | Iterable[PolarsExprType | PythonLiteral]
+            | None
+        ) = None,
         **named_exprs: Any,
```

Comment:
Why are these `Any`? They can be type hinted as well, right?

```suggestion
        **named_exprs: PolarsExprType | PythonLiteral,
```

Reply:
Precisely because specifying all the individual types was becoming increasingly ludicrous, heh. Can now rationalise as suggested :)